### ISSUE TITLE
yield-inside-async-function false positive in Python 3.6

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#1372]
Given this code:

```
async def ticker(delay, to):
    """Yield numbers from 0 to *to* every *delay* seconds."""
    for i in range(to):
        yield i
        await asyncio.sleep(delay)
```
pylint emits a ``yield-inside-async-function`` error. It seems that since Python 3.6 this is allowed.
